manifests/osfamily/redhat.pp

```diff
@@ -88,14 +88,16 @@
     logoutput => 'on_failure',
   }
 
-  yumrepo { 'pc_repo':
-    baseurl       => $source,
-    descr         => "Puppet Labs ${::puppet_agent::collection} Repository",
-    enabled       => true,
-    gpgcheck      => '1',
-    gpgkey        => "${gpg_keys}",
-    sslcacert     => $_sslcacert_path,
-    sslclientcert => $_sslclientcert_path,
-    sslclientkey  => $_sslclientkey_path,
+  if $::puppet_agent::manage_repo {
```

Comment:
For my particular use case, I thought it was beneficial to still have puppet_agent install the gpg keys so there's no need to distribute them through another mechanism and the rpms could be loaded in internal repos without being re-signed.

That said, I could see the argument being made that if the ability to disable repo creation is there, then the same ability should exist for the gpg key.